src/System.Device.Gpio/System/Device/I2c/I2cDevice.cs

```diff
@@ -12,6 +14,30 @@ public abstract class I2cDevice : IDisposable
         public abstract void WriteByte(byte data);
         public abstract void Write(Span<byte> data);
 
+        /// <summary>
+        /// Initializes new instance of I2cDevice based on OS platform.
+        /// </summary>
+        /// <param name="busId">The bus ID the device is connected to.</param>
+        /// <param name="deviceAddress">The bus address of the device.</param>
+        public static I2cDevice CreateI2cDevice(int busId, int deviceAddress)
```

Comment:
I believe that what we want here instead of this is to have a Controller concept, similar to Gpio and PWM, where the controller manages one or more I2c devices, and it is the one in charge of picking either the Linux or Windows version of it.

Reply:
So basically have something like a I2cController class that hasn't been added to repo yet?  If so, I'm assuming you could also pass in the specific device in overloaded constructor like I2cController(I2cDevice i2cDevice)?  This makes a little more sense when it comes to bindings.

Reply:
Right, we were thinking of adding the I2c and Spi controller classes for the next version, whose job would be to provide the right device for a given runtime.